From 2949ceab1ea4618fcac5668a99f89dcb272c3875 Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Sat, 17 May 2025 18:01:33 +0000
Subject: [PATCH] build: replace openssl with libcrypto in pkgconfig

Signed-off-by: Ben Cressey <bcressey@amazon.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6a6c4df..a00dcc7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -717,7 +717,7 @@ dnl pwquality is used only by tools
 PKGMODULES="uuid devmapper json-c"
 case $with_crypto_backend in
 	gcrypt)  PKGMODULES="$PKGMODULES libgcrypt" ;;
-	openssl) PKGMODULES="$PKGMODULES openssl" ;;
+	openssl) PKGMODULES="$PKGMODULES libcrypto" ;;
 	nss)     PKGMODULES="$PKGMODULES nss" ;;
 	nettle)  PKGMODULES="$PKGMODULES nettle" ;;
 esac
